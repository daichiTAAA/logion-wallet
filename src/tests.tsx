import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
