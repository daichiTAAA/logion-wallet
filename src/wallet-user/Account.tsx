import React from 'react';

import { ContentPane } from '../common/Dashboard';
import Frame from '../common/Frame';

import { useCommonContext } from "../common/CommonContext";
import MyTokens from './MyTokens';
import Tokenization from "./Tokenization";
import RefreshRequestsButton from './RefreshRequestsButton';
import PendingTokenizationRequests from './PendingTokenizationRequests';
import AcceptedTokenizationRequests from './AcceptedTokenizationRequests';
import RejectedTokenizationRequests from './RejectedTokenizationRequests';

export default function Account() {
    const { colorTheme } = useCommonContext();

    return (
        <ContentPane
            mainTitle="Tokens"
            titleIcon={{
                icon: {
                    id: 'tokens'
                },
                background: colorTheme.topMenuItems.iconGradient,
            }}
            primaryAreaChildren={
                <>
                    <Frame>
                        <MyTokens />
                        <Tokenization/>
                        <RefreshRequestsButton/>
                        <PendingTokenizationRequests />
                    </Frame>
                </>
            }
            secondaryAreaChildren={
                <>
                    <Frame>
                        <h2>History</h2>
                        <AcceptedTokenizationRequests />
                        <RejectedTokenizationRequests />
                    </Frame>
                </>
            }
        />
    );
}
