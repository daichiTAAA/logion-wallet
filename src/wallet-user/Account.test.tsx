jest.mock('../common/RootContext');

import React from "react";
import {shallowRender} from "../tests";
import Account from "./Account";

test("renders", () => {
    const tree = shallowRender(<Account/>)
    expect(tree).toMatchSnapshot();
});
