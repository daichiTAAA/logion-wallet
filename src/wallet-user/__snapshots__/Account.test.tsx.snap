// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ContentPane
  primaryAreaChildren={
    <React.Fragment>
      <h1>
        Tokens
      </h1>
      <Frame
        colors={
          Object {
            "accounts": Object {
              "background": "#0c163d",
              "foreground": "#ffffff",
              "hintColor": "#ffffff99",
              "iconBackground": "#3b6cf4",
              "legalOfficerIcon": Object {
                "category": "legal-officer",
                "id": "account-shield",
              },
            },
            "bottomMenuItems": Object {
              "iconGradient": Object {
                "from": "#7a90cb",
                "to": "#3b6cf4",
              },
            },
            "buttons": Object {
              "secondaryBackgroundColor": "#0c163d",
            },
            "dashboard": Object {
              "background": "#0c163d",
              "foreground": "#ffffff",
            },
            "frame": Object {
              "background": "#152665",
              "foreground": "#ffffff",
              "link": "#ffffff",
            },
            "shadowColor": "#3b6cf433",
            "sidebar": Object {
              "activeItemBackground": "#ffffff0f",
              "background": "#152665",
              "foreground": "#ffffff",
            },
            "topMenuItems": Object {
              "iconGradient": Object {
                "from": "#3b6cf4",
                "to": "#6050dc",
              },
            },
            "type": "dark",
          }
        }
      >
        <MyTokens />
        <Tokenization />
        <RefreshRequestsButton />
        <PendingTokenizationRequests />
      </Frame>
    </React.Fragment>
  }
  secondaryAreaChildren={
    <React.Fragment>
      <Frame
        colors={
          Object {
            "accounts": Object {
              "background": "#0c163d",
              "foreground": "#ffffff",
              "hintColor": "#ffffff99",
              "iconBackground": "#3b6cf4",
              "legalOfficerIcon": Object {
                "category": "legal-officer",
                "id": "account-shield",
              },
            },
            "bottomMenuItems": Object {
              "iconGradient": Object {
                "from": "#7a90cb",
                "to": "#3b6cf4",
              },
            },
            "buttons": Object {
              "secondaryBackgroundColor": "#0c163d",
            },
            "dashboard": Object {
              "background": "#0c163d",
              "foreground": "#ffffff",
            },
            "frame": Object {
              "background": "#152665",
              "foreground": "#ffffff",
              "link": "#ffffff",
            },
            "shadowColor": "#3b6cf433",
            "sidebar": Object {
              "activeItemBackground": "#ffffff0f",
              "background": "#152665",
              "foreground": "#ffffff",
            },
            "topMenuItems": Object {
              "iconGradient": Object {
                "from": "#3b6cf4",
                "to": "#6050dc",
              },
            },
            "type": "dark",
          }
        }
      >
        <h2>
          History
        </h2>
        <AcceptedTokenizationRequests />
        <RejectedTokenizationRequests />
      </Frame>
    </React.Fragment>
  }
/>
`;
